Extension/src/LanguageServer/lmTool.ts

```diff
@@ -83,25 +74,33 @@ export class CppConfigurationLanguageModelTool implements vscode.LanguageModelTo
             let contextString = "";
             if (chatContext.language) {
                 contextString += `The user is working on a ${chatContext.language} project. `;
+                telemetryProperties["language"] = chatContext.language;
             }
             if (chatContext.standardVersion) {
                 contextString += `The project uses language version ${chatContext.standardVersion}. `;
+                telemetryProperties["standardVersion"] = chatContext.standardVersion;
             }
             if (chatContext.compiler) {
                 contextString += `The project compiles using the ${chatContext.compiler} compiler. `;
+                telemetryProperties["compiler"] = chatContext.compiler;
             }
             if (chatContext.targetPlatform) {
                 contextString += `The project targets the ${chatContext.targetPlatform} platform. `;
+                telemetryProperties["targetPlatform"] = chatContext.targetPlatform;
             }
             if (chatContext.targetArchitecture) {
                 contextString += `The project targets the ${chatContext.targetArchitecture} architecture. `;
+                telemetryProperties["targetArchitecture"] = chatContext.targetArchitecture;

```

Comment:
note that this is a "breaking"  change for the values reported in telemetry (differences in casing but more importantly also in syntax, e.g. `cpp` becomes now `C++`)

Reply:
Acknowledged. I'm comfortable saying that's an acceptable break given that we're not currently querying this telemetry, and it will bring the telemetry closer to what the user actually experiences. It will also give us a better signal if data is ever missing.

Reply:
casing of events and properties is not respected in the telemetry system. They are stored as "lower-case." Only the casing of values is preserved.